### ISSUE TITLE
Support NC24 (and thus support PHP 8.1)

### ISSUE THREAD

**Issue by nursoda** [nextcloud/mail#6355]
### Is your feature request related to a problem? Please describe.

On an Arch setup (NC 24RC3, PHP 8.1.5), I cannot (re)activate the Mail app:
`PHP with a version lower than 8.0 is required.`

### Describe the solution you'd like

Support latest release versions, especially NC ([due tomorrow](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule)) and PHP (released in [November 2021](https://www.php.net/archive/2021.php#2021-11-25-1)).

### Describe alternatives you've considered

Disable Mail? Not upgrade to NC24? Use another Mail app? Well…

**Comment by ChristophWurst** [comment]
Use https://github.com/nextcloud-releases/mail/releases/tag/v1.12.0-rc3 if you also use the RC of server or wait until tomorrow for the final.

For PHP8.1 I invite you to read through https://github.com/nextcloud/mail/pull/6131 and give us your best bet on why PHP errors out.

**Comment by nursoda** [comment]
Sorry for the unnecessary report. ( I honestly did look for (just) issues related to NC24 and/or PHP 8.1 but didn't find any. I also looked into the first page of PR's but didn't search there. )

Concerning your request, unfortunately I'm not a dev, I just _tried_ to help. As I understand the PR automation error, your unit test fails. I don't know anything about how to set up a proper PHP test environment (xdebug? phpunit?) or how to write or debug tests, so I'm of no help here.

In case it's relevant: After upgrading my instances to NC24 I saw many parallel PHP coredumps in htop. After they crashed the server altogether, I disabled NC apps I didn't need much, Mail among them. The coredumps stopped, so I *assume* there is a connection. I can provide them (but would need instructions) or let you analyze them on my server if that's quicker.
